scripts/ca/nb/geonb.sh

```diff
@@ -0,0 +1,142 @@
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/coordinate_sources.txt > $TMP/geonb_gcadb-bdavg_csv/coordinate_sources.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/counties.txt > $TMP/geonb_gcadb-bdavg_csv/counties.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/locations.txt > $TMP/geonb_gcadb-bdavg_csv/locations.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/place_names.txt > $TMP/geonb_gcadb-bdavg_csv/place_names.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/street_directions.txt > $TMP/geonb_gcadb-bdavg_csv/street_directions.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/street_types.txt > $TMP/geonb_gcadb-bdavg_csv/street_types.txtutf
+iconv -f latin1 -t utf-8//TRANSLIT $TMP/geonb_gcadb-bdavg_csv/unit_types.txt > $TMP/geonb_gcadb-bdavg_csv/unit_types.txtutf
+
+echo "
+    update pg_database set encoding = pg_char_to_encoding('UTF8') where datname = 'geonb';
+" | psql -U postgres
+
+echo "
+    COPY coordinate_classes FROM '$TMP/geonb_gcadb-bdavg_csv/coordinate_classes.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY coordinate_methods FROM '$TMP/geonb_gcadb-bdavg_csv/coordinate_methods.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY coordinate_sources FROM '$TMP/geonb_gcadb-bdavg_csv/coordinate_sources.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY counties FROM '$TMP/geonb_gcadb-bdavg_csv/counties.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY locations FROM '$TMP/geonb_gcadb-bdavg_csv/locations.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY place_names FROM '$TMP/geonb_gcadb-bdavg_csv/place_names.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY street_directions FROM '$TMP/geonb_gcadb-bdavg_csv/street_directions.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY street_types FROM '$TMP/geonb_gcadb-bdavg_csv/street_types.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+    COPY unit_types FROM '$TMP/geonb_gcadb-bdavg_csv/unit_types.txtutf' HEADER CSV DELIMITER E'\t' QUOTE E'\b' ENCODING 'UTF8';
+" | psql -U postgres geonb
+
+echo "
+    DELETE FROM locations WHERE civic_num IS NULL OR civic_num = '0';
+" | psql -U postgres
+
+echo "
+    CREATE TABLE final AS
+        SELECT
+            l.id AS id,
+            l.pi_par_pid AS  pid,
+            l.civic_num||COALESCE(l.civic_num_suff,'') AS number,
+            CASE
+                WHEN street_types.lang_cd = 1 THEN
+                    street_name||COALESCE(' '||street_types.name, '')||COALESCE(' '||street_directions.name_e, '')
+                ELSE
+                    COALESCE(street_types.name,'')||street_name||COALESCE(street_directions.name_f, '')
```

Comment:
Noticed some of the French street names in the new file are missing a space e.g. "RueVeniot".

One idea that might make life easier in these sorts of situations is using the concat_ws function instead of COALESCE. Does the same thing (including ignoring NULL arguments) but with nicer syntax. Example:

```
concat_ws(' ', street_types.name, street_name, street_directions.name_f)
```